src/Aspire.Hosting/ApplicationModel/EndpointAnnotation.cs

```diff
@@ -31,7 +35,19 @@ public sealed class EndpointAnnotation : IResourceAnnotation
     /// <param name="targetPort">This is the port the resource is listening on. If the endpoint is used for the container, it is the container port.</param>
     /// <param name="isExternal">Indicates that this endpoint should be exposed externally at publish time.</param>
     /// <param name="isProxied">Specifies if the endpoint will be proxied by DCP. Defaults to true.</param>
-    public EndpointAnnotation(ProtocolType protocol, string? uriScheme = null, string? transport = null, [EndpointName] string? name = null, int? port = null, int? targetPort = null, bool? isExternal = null, bool isProxied = true)
+    /// <param name="networkID">The ID of the network that is the "default" network for the Endpoint.
+    /// Clients connected to the same network can reach the endpoint without any routing or network address translation.</param>
+    public EndpointAnnotation(
+        ProtocolType protocol,
+        string? uriScheme = null,
+        string? transport = null,
+        [EndpointName] string? name = null,
+        int? port = null,
+        int? targetPort = null,
+        bool? isExternal = null,
+        bool isProxied = true,
+        NetworkIdentifier? networkID = null
```

Comment:
This is a breaking API chang still, versioning ctors like this is a pain. We need to make a new overload with all of the old parameters fixed so people using the old API won't get method missing.